packages/client/src/internals/TRPCClient.ts

```diff
@@ -215,7 +218,7 @@ export class TRPCClient<TRouter extends AnyRouter> {
       TRPCRequestOptions?,
     ]
   ) {
-    const context = (args[1] as TRPCRequestOptions | undefined)?.requestContext;
```

Comment:
Why are we spreading the inferHandlerInput and then typecasting TRPCRequestOptions as the 2nd element? Wouldn't this be cleaner, or is there something going on under the hood that makes this mandatory?
```ts
public query<
    TQueries extends AssertLegacyDef<TRouter>['queries'],
    TPath extends string & keyof TQueries,
  >(
    path: TPath,
    ...args: [
       inferHandlerInput<AssertType<TQueries, ProcedureRecord>[TPath]>[0],
       TRPCRequestOptions?,
     ]
   ) {
    const context = args[1]?.requestContext;
    //    ^? TRPCRequestOptions | undefined
```

Reply:
A lot of this is very old stuff so might not be a good reason

Reply:
Good catch, I think this is actually a mistake. It allows you to do this:
```ts
client.query(
  "greeting",
  { text: "hello" },
  {
    context: {
      batch: false,
    },
  },
  {
    context: {
      batch: false,
    },
  }
);
```
pass two options objects and the last one will just be ignored.

Reply:
I can fix this in a separate PR